pkg/netstore/netstore.go

```diff
@@ -51,13 +73,34 @@ func (s *store) Get(ctx context.Context, mode storage.ModeGet, addr swarm.Addres
 				if err != nil {
 					return nil, err
 				}
-				go func() {
-					err := s.recoveryCallback(addr, targets)
-					if err != nil {
-						s.logger.Debugf("netstore: error while recovering chunk: %v", err)
+				chunkC := make(chan swarm.Chunk, 1)
+				socAddress, err := s.recoveryCallback(addr, targets, chunkC)
+				if err != nil {
+					s.logger.Debugf("netstore: error while recovering chunk: %v", err)
```

Comment:
The error is just logged and execution of the function continues, but it should not. Error should be returned.